setuptools/glob.py

```diff
@@ -129,8 +127,8 @@ def glob2(dirname, pattern):
 # Recursively yields relative pathnames inside a literal directory.
 def _rlistdir(dirname):
     if not dirname:
-        if isinstance(dirname, binary_type):
-            dirname = binary_type(os.curdir, 'ASCII')
+        if isinstance(dirname, bytes):
+            dirname = os.curdir.decode('ASCII')
```

Comment:
Again, this seems like a change to do more than just replace binary_type with bytes. How did this code work before?